packages/unchained-client/src/evm/parser/erc20.ts

```diff
@@ -45,7 +55,8 @@ export class Parser<T extends Tx> implements SubParser<T> {
           assetReference: tx.to
         }),
         method: decoded.name,
-        parser: TxParser.ERC20
+        parser: TxParser.ERC20,
+        ...(approveValue ? { value: approveValue } : {})
```

Comment:
Notes:
- Not sure if we should stick to the actual argument name e.g `value` for yearn, `amount` for ERC20s, matching the `_amount` arg
- Not sure if this is the best place / types for this? We might have other functions we support that might take an amount parameter, or other parameters from other functions we might want to parse as well. Maybe `args` or `arguments` as a `Record<T, U>` might be a better fit?

Reply:
I say stick with `value` naming just so you can expect to see the similar data across the whole parsed tx payload imo.

A record would certainly be easier, but it would make consumption difficult client side as you have no idea what values could be in there. Depending on how complex these get, there is always the possibility of adding a contractArgs object that could be defined based on the contract method as a discriminator, but that is getting really close to just turning into ethersjs/web3 abi parsing at that point. Could be useful to pass back the `inputData` as an alternative for more custom needs as well